extension/src/cli/git/reader.ts

```diff
@@ -31,18 +31,28 @@ export class GitReader extends GitCli {
     return !!output
   }
 
-  public async getCommitMessage(cwd: string, sha: string) {
+  public async getLastThreeCommitMessages(cwd: string): Promise<{
+    [sha: string]: string
+  }> {
     const options = getOptions(
       cwd,
       Command.LOG,
       Flag.PRETTY_FORMAT_COMMIT_MESSAGE,
       Flag.NUMBER,
-      '1',
-      sha
+      '3',
+      Flag.SEPARATE_WITH_NULL
     )
     try {
-      return await this.executeProcess(options)
-    } catch {}
+      const output = await this.executeProcess(options)
```

Comment:
I would keep this code as dumb as possible. Return a string and let `collectExperiments` deal with it. We want to try and keep all of the parsing logic in one place.

Reply:
Parsing the string somewhere in `addCommitDataToBranches` would make sense to me

Reply:
Would be good to add a test for the parsing logic after it gets moved too.